wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -295,6 +302,83 @@ public static function generateCollectionCampQr(string $op, string $objectName,
       return;
     }
 
+    $collectionCamps = EckEntity::get('Collection_Camp', TRUE)
+      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id')
+      ->addWhere('id', '=', $objectId)
+      ->execute();
+
+    $currentCollectionCamp = $collectionCamps->first();
+    $currentStatus = $currentCollectionCamp['Collection_Camp_Core_Details.Status'];
+    $collectionCampId = $currentCollectionCamp['id'];
+    $contactId = $currentCollectionCamp['Collection_Camp_Core_Details.Contact_Id'];
+
+    // Check for status change.
+    if ($currentStatus !== $newStatus) {
+      if ($newStatus === 'authorized') {
+        self::generateQrCode($contactId, $collectionCampId);
+      }
+    }
+  }
+
+  public static function generateQrCode($contactId, $collectionCampId) {
+    try {
+      $baseUrl = CRM_Core_Config::singleton()->userFrameworkBaseURL;
+      $url = "{$baseUrl}actions/collection-camp/{$collectionCampId}";
+  
+      $options = new QROptions([
+        'version'    => 5,
+        'outputType' => QRCode::OUTPUT_IMAGE_PNG,
+        'eccLevel'   => QRCode::ECC_L,
+        'scale'      => 10,
+      ]);
+  
+      $qrcode = (new QRCode($options))->render($url);
+  
+      // Remove the base64 header and decode the image data.
+      $qrcode = str_replace('data:image/png;base64,', '', $qrcode);
+
+      $qrcode = base64_decode($qrcode);
+  
+      $baseFileName = "qr_code_{$contactId}.png";
+
+      $fileName = CRM_Utils_File::makeFileName($baseFileName);
+
+      $tempFilePath = CRM_Utils_File::tempnam($baseFileName);
+
+      $numBytes = file_put_contents($tempFilePath, $qrcode);
+
+      if (!$numBytes) {
+        CRM_Core_Error::debug_log_message('Failed to write QR code to temporary file for contact ID ' . $contactId);
+        return FALSE;
+      }
+
+      // Save the QR code as an attachment linked to the contact.
+      $params = [
+        'entity_id' => $contactId,
+        'name' => $fileName,
+        'mime_type' => 'image/png',
+        'field_name' => 'custom_215',
```

Comment:
Create a new custom field group called: `Collection Camp QR Code`

This group will have have only one field called: `QR Code`.

Please refer to the contact custom field